src/crypto/crypto_sig.cc

```diff
@@ -829,56 +673,53 @@ bool SignTraits::DeriveBits(
     Environment* env,
     const SignConfiguration& params,
     ByteSource* out) {
+  ClearErrorOnReturn clear_error_on_return;
   EVPMDPointer context(EVP_MD_CTX_new());
   EVP_PKEY_CTX* ctx = nullptr;
 
-  ManagedEVPPKey m_pkey = params.key->GetAsymmetricKey();
-  Mutex::ScopedLock lock(*m_pkey.mutex());
```

Comment:
@danbev this mutex here drastically affects parallel performance in crypto.sign and crypto.verify with KeyObject as input. Could you please look at the code affected by its removal and suggest where it needs to be re-added without it affecting parallel sign/verify execution this much?

```js
crypto/oneshot-sign-verify.jsn=1000 keyFormat='keyObject' mode='async-parallel'               ***    233.22 %       ±3.06% ±4.05% ±5.22%
```

Reply:
I cannot say by just looking at the code. I'd have to run this and link it against OpenSSL 3.0 to be sure that the code is thread safe when moving the lock. This is not noticed when linking with OpenSSL 1.1.1. 

Alternatively there will be a [CI job](https://github.com/nodejs/build/issues/2584) created that will link against OpenSSL 3.0 and once that is in place one should be able to run the CI tests and be able to verify that changes don't break. Would it be possible to put this PR on hold until that CI jobs is in place?

Reply:
> I'd have to run this and link it against OpenSSL 3.0 to be sure that the code is thread safe when moving the lock.

That's what I was hoping for actually.

KeyObject use is meant to always be beneficial but in this particular case you get better throughput when using pem as input which is unexpected to say the least.

Reply:
CI job is in and passed even with an added libuv threaded test.